### PR TITLE
Tajara Biology Additions: Retro Biopunk Edition

### PR DESCRIPTION
**Pull request by Alberyk**
This pr changes some stuff with Tajaran biology.
-tajara have a higher nutrition and hydration thresholds and get hungry slower, this is a bit different between subspecies
-tajara process chemicals a bit slower
-tajara have their own set of organs with some differences in functions, flavor and icons
-tajara now have a somewhat proper night vision verb that can be toggled, while it allows the user to see in the darkness, it removes all the color. Being flashed or hurt in the eyes disables it. Tajaran eyes are also bigger and thus have a higher chance of being hit when attacking the head.

### CODE REVIEW

**Comment by KasparoVy** [comment]
Awesome.

File: code/modules/organs/internal/species/tajara.dm [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+/obj/item/organ/internal/eyes/night/proc/enable_night_vision()
+	if(!owner)
+		return
+	if(night_vision)
+		return
+	var/show_message = TRUE
+	for(var/obj/item/protection in list(owner.head, owner.wear_mask, owner.glasses))
+		if((protection && (protection.body_parts_covered & EYES)))
+			show_message = FALSE
```
Comment (Geevies) [inline code/modules/organs/internal/species/tajara.dm:74]: could add a break below here to terminate the for if we find a suitable thing

File: code/modules/organs/internal/species/tajara.dm [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+/obj/item/organ/internal/eyes/night/take_internal_damage(var/amount, var/silent = 0)
+	..()
+	disable_night_vision()
+
+/obj/item/organ/internal/eyes/night/flash_act()
+	if(!owner)
+		return
+
+	to_chat(owner, "<span class='warning'>Your eyes burn with the intense light of the flash!</span>")
```
Comment (Geevies) [inline code/modules/organs/internal/species/tajara.dm:61]:
```suggestion
	to_chat(owner, SPAN_WARNING("Your eyes burn with the intense light of the flash!"))
```

**Review by Geevies — APPROVED** [review]

File: code/modules/organs/internal/species/tajara.dm [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	if(!owner)
+		return
+	if(night_vision)
+		return
+	var/show_message = TRUE
+	for(var/obj/item/protection in list(owner.head, owner.wear_mask, owner.glasses))
+		if((protection && (protection.body_parts_covered & EYES)))
+			show_message = FALSE
+	if(show_message)
+		owner.visible_message(SPAN_NOTICE("\The [owner]'s eyes dilate!"))
```
Comment (Geevies) [inline code/modules/organs/internal/species/tajara.dm:76]: a var on the eyes to adjust this message without proc copy paste would be nice

File: code/modules/organs/internal/species/tajara.dm [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+	night_vision = TRUE
+	owner.stop_sight_update = TRUE
+	owner.see_invisible = SEE_INVISIBLE_NOLIGHTING
+	owner.add_client_color(/datum/client_color/monochrome)
```
Comment (Geevies) [inline code/modules/organs/internal/species/tajara.dm:81]: same here for the client_color type

File: code/modules/organs/internal/species/tajara.dm [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+		if(action.button)
+			action.button.update_icon()
+
+/obj/item/organ/internal/eyes/night/attack_self(var/mob/user)
+	. = ..()
+	if(owner.last_special > world.time)
+		return
+
+	if(owner.stat || owner.paralysis || owner.stunned || owner.weakened)
+		return
```
Comment (NonQueueingMatt) [inline code/modules/organs/internal/species/tajara.dm:36]: use_check with USE_ALLOW_NON_ADJACENT should work here?

File: code/modules/organs/internal/species/tajara.dm [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+	if(is_broken())
+		return
+
+	if((status & ORGAN_ROBOT))
```
Comment (NonQueueingMatt) [inline code/modules/organs/internal/species/tajara.dm:41]: what's this for and why? if this is fine, the double parenthesis are useless

**Review by NonQueueingMatt — APPROVED** [review]

File: code/modules/organs/internal/species/tajara.dm (hunk 6 above)
Comment (Alberyk) [reply code/modules/organs/internal/species/tajara.dm:41]: This is to stop robotic eyes from using this skill.

**Pull request merged**